src/main/java/listeners/OnlinerTestListener.java

```diff
@@ -39,19 +42,24 @@ public void onTestFailedWithTimeout(ITestResult result) {
     @Override
     public void onStart(ITestContext context) {
         logger.info("The test class run started on " + context.getStartDate());
-        allureEnvironmentWriter(
-                ImmutableMap.<String, String>builder()
-                        .put("Browser", "Chrome")
-                        .put("Browser.Version", "96.0.4664.93 (Official build), (64 bit)")
-                        .build(), System.getProperty("user.dir")
-                        + "/target/allure-results/");
+        context.getCurrentXmlTest().getSuite().setParallel(XmlSuite.ParallelMode.METHODS);
+        context.getCurrentXmlTest().getSuite().setThreadCount(ThreadCountConfig.getThreads());
     }
 
     @Override
     public void onFinish(ITestContext context) {
         logger.info("The test class run finished on ... " + context.getEndDate());
     }
 
+    public static void attachEnvironmentInfo(WebDriver driver ) {
+        allureEnvironmentWriter(
+                ImmutableMap.<String, String>builder()
+                        .put("Browser", ((RemoteWebDriver) driver).getCapabilities().getBrowserName())
```

Comment:
Why do you use RemoteWebDriver? can't you get capabilities from a WebDriver?

Reply:
from the WebDriver - no. From drivers which inherit from RemoteDriver - yes (ChromeDriver, EdgeDriver etc)